libraries/rhel_handler.rb

```diff
@@ -52,16 +56,15 @@ def configure_package(action_name)
           action :delete
           only_if { ::File.exist?('/etc/yum.repos.d/chef_stable_.repo') }
         end
-
         # Enable the required yum-repository.
-        include_recipe "yum-chef::#{new_resource.channel}"
+        include_recipe "yum-chef::#{new_resource.channel}" unless byop
 
         # Foodcritic doesn't like timeout attribute in package resource
         package new_resource.product_name do # ~FC009
           action action_name
           package_name ingredient_package_name
           # Ensure that we are installing from the correct repository
-          options "--disablerepo=* --enablerepo=chef-#{new_resource.channel} #{new_resource.options}"
+          options "--disablerepo=* --enablerepo=chef-#{new_resource.channel} #{new_resource.options}" unless byop
```

Comment:
this is still needed even with byop. maybe byop could contain the name of the yum repo to use and something like `--enablerepo=#{byop}` could be used here